docs/pages/server-access/guides/host-user-creation.mdx

```diff
@@ -0,0 +1,153 @@
+---
+title: Configure Teleport to Create Host Users
+description: How to configure Teleport to automatically create transient host users.
+---
+
+Teleport's SSH Service can be configured to automatically create local Unix users
+upon login.
+
+Host users created by Teleport are transient and will be deleted at the end of an SSH
+session. This feature has overlap with the [PAM](./ssh-pam.mdx) integration, however
+it provides some additional convenience of not having to manually configure a PAM
+script to create and delete users.
+
+It also has benefits over manually creating users as it allows for separate host
+users for each member of an organization. This provides more fine-grained control
+of permissions on a given host.
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- A running Teleport Node. See the [Server Access Getting Started Guide](../getting-started.mdx) for  how to add a Node to your Teleport cluster.
+- The following utilities should be available in the PATH for the Teleport SSH Service,
+since it must execute these commands in order to create transient users:
+  - `useradd`
+  - `userdel`
+  - `usermod`
+  - `groupadd`
+  - `getent`
+  - `visudo`
+
+(!docs/pages/includes/tctl.mdx!)
+
+## Step 1/2. Configure a role
+
+First, create a role with `create_host_user` set to `true`. This will allow users
+with this role to have transient host users created at login time.
+
+The following role specification will allow users to log in as `nginxrestarter` on
+any matching Node. The host user will be created and added to the groups listed in
+`host_groups`. They will also be given permission to restart the Nginx service as
+root.
+
+Save the file below as `auto-users.yaml`
+
+```yaml
+kind: role
+version: v5
+metadata:
+  name: auto-users
+spec:
+  options:
+    # Allow automatic creation of users.
+    create_host_user: true
+  allow:
+    logins: [ "nginxrestarter" ]
+    # List of host groups the created user will be added to. Any that don't already exist are created.
+    host_groups: [ubuntu, nginx, other]
+    # List of entries to include in a temporary sudoers file created in /etc/sudoers.d
+    host_sudoers: [
+       # This line will allow the `nginxrestarter` user to run
+       # `systemctl restart nginx.service` as
+       # root without requiring a password.
+       #
+       # sudoers file reference documentation: https://www.sudo.ws/docs/man/1.8.17/sudoers.man/
+       "{{internal.logins}} ALL = (root) NOPASSWD: /usr/bin/systemctl restart nginx.service"
```

Comment:
Left a comment here with a suggestion: https://github.com/gravitational/teleport/issues/13663#issuecomment-1170232352.

I think we need to introduce a `{{internal.host_login}}` variable that will expand to the current host username and use it in the guide as well instead of "logins" trait.